src/ui/components/Avatar.js

```diff
@@ -1,20 +1,43 @@
-import React, { useState, useEffect } from "react";
+import React, { useEffect } from "react";
 import { useAuth0 } from "@auth0/auth0-react";
+import { gql, useMutation } from "@apollo/client";
 
 import { connect } from "react-redux";
 import { actions } from "ui/actions";
 import { getAvatarColor } from "ui/utils/user";
 import { setUserInBrowserPrefs } from "../utils/browser";
+import { ThreadFront } from "protocol/thread";
 
+const CREATE_SESSION = gql`
+  mutation AddSession($object: sessions_insert_input!) {
+    insert_sessions_one(object: $object) {
+      id
+      controller_id
+      recording_id
+    }
+  }
+`;
 const Avatar = props => {
   let { player, isFirstPlayer, updateUser } = props;
   let auth = useAuth0();
+  const [AddSession] = useMutation(CREATE_SESSION);
 
   useEffect(() => {
     setUserInBrowserPrefs(auth.user);
     updateUser(auth.user);
   }, [auth.user]);
 
+  useEffect(() => {
+    if (auth.user?.sub && ThreadFront.sessionId) {
+      const object = {
+        id: ThreadFront.sessionId,
+        recording_id: ThreadFront.recordingId,
+        controller_id: ThreadFront.sessionId.split("/")[0],
+      };
+      AddSession({ variables: { object } });
+    }
+  }, [auth.user?.sub]);
```

Comment:
I think we can just subscribe to changes to `auth.user` instead of `auth.user.sub` here, since once we get the user from auth0, their `sub` doesn't change.

Reply:
yeah that's fair